### PR TITLE
CM-566: User can reset his forgotten password (Part 1)

### PR DESCRIPTION
**Pull request by Svenstar74**
We will split this ticket into two different PRs. This one will:
- Provide the user a button + dialog to reset his password
- Send the email to the user

The next PR then continues, when the user opens the link in the email to set his new password. We do this to better test the sendgrid email stuff (and the PR would probably become quite large and chaotic otherwise as well).

image

### CODE REVIEW

**Review by rodriguesk — CHANGES_REQUESTED** [review]
Looks great (in terms of what the screens look like). I think the button used to submit the email should say "submit" instead of "confirm". @NickCallaghan will have to approve the rest of the story regarding tests, style, and actual code.

**Review by rodriguesk — APPROVED** [review]

File: src/api/postPasswordResetLink.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  const url = '/password-reset';
+
+  try {
+    const request = await climateApi({
+      method: 'post',
+      url,
+      data: {
+        email,
+      },
+      withCredentials: true,
```
Comment (NickCallaghan) [inline src/api/postPasswordResetLink.ts:23]: are credentials needed? If the user has forgotten their password will they not be logged out?

Comment (Svenstar74) [reply src/api/postPasswordResetLink.ts]: Removed the withCredentials part

**Review by NickCallaghan — DISMISSED** [review]
Looks great @Svenstar74 Thanks for adding the tests. Looks like there is a conflict to resolve but feel free to merge after that is solved.

**Review by rodriguesk — APPROVED** [review]
Awesome job!

**Comment by rodriguesk** [comment]
@Svenstar74 anyhold up on merging this to develop? If it’s ready please merge it yourself :)

**Pull request merged**